_posts/2016-10-25-cve-2016-5195-dirtycow-mitigated.md

```diff
@@ -0,0 +1,47 @@
+* Various drivers, which are also probably using the flag by mistake.
+
+As it turns out, none of these code paths can be exploited by Sandstorm apps:
+
+* Sandstorm uses seccomp to block the app from invoking `ptrace()`.
+* Sandstorm does not mount `/proc` inside app sandboxes.
+* Sandstorm does not expose driver interfaces from inside app sandboxes. For example, `/dev` contains only `null`, `zero`, `urandom`, and `random` (which is actually a synonym for `urandom`, [as it](https://gist.github.com/tarcieri/6347417) [should be](http://www.2uo.de/myths-about-urandom/)).
+
+So, as far as we can tell, Sandstorm has never been vulnerable to this bug.
+
+### Defense in depth
+
+Even if Sandstorm were vulnerable, the exploit would have far reduced impact inside Sandstorm than in a typical Linux environment, because:
+
+* A Sandstorm app's filesystem consists of the contents of its own package. It cannot see the host system's files nor files belonging to other apps, hence it would not be able to memory-map them in order to modify them using this bug.
+* App packages cannot contain setuid binaries and, even if they could, apps would not be able to execute them, because Sandstorm sets the `NO_NEW_PRIVS` `prctl()` flag inside the sandbox.
+
+Because of these considerations, even if Sandstorm had not mitigated the bug outright, the impact of the bug would be that one instance of an app could attack other instances of the same app (running on the same machine in separate containers). Most apps do not allow a remote user to upload arbitrary code to execute and, hence, would only be vulnerable if the app author had included attack code in the app. On Sandstorm, we would consider this a vulnerability, because Sandstorm seeks to implement "confinement", meaning that an app should not be able to "phone home" to its author or speak to other copies of itself without permission.
+
+Interestingly, though, on most platforms, it's fully expected that instances of the same app can voluntarily collude with each other, e.g. by communicating with a central coordinator or by simply opening network connections to each other. And actually, Sandstorm does not yet fully implement confinement either -- we intend to, and an app has to go out of its way to break confinement today, but we have not yet prioritized systematically closing all holes.
+
+So, even if Sandstorm had not directly mitigated this bug, and barring some as-yet-unreported impact of this vulnerability, we believe that it would not have allowed the app to do anything except coordinate with other instances of itself, which it could already do by communicating through the network, which on most platforms wouldn't even be considered a bug.
+
+### The List
+
+This is not the first Linux security bug mitigated by Sandstorm. In fact, we've kept a long list. Check it out at: [Sandstorm Security Non-Events](https://docs.sandstorm.io/en/latest/using/security-non-events/#linux-kernel)
```

Comment:
I'd love it if the article would end with some kind of call to action about what's interesting about Sandstorm here. Perhaps the fact that we also mitigate app bugs - this fact is fairly shocking to people - followed by a suggestion that you can try Sandstorm and see that it's also easy to use.